tests/rbd/test_rbd_encryption.py

```diff
@@ -0,0 +1,163 @@
+            log.error(f"Clone creation failed for {clone_spec}")
+
+        log.info(
+            f"Applying encryption of type {encryption_type} on the clone {clone_spec}"
+        )
+        clone_passphrase = "clone_luks1_passphrase.bin"
+        rbd.create_passphrase_file(clone_passphrase)
+        if rbd.encrypt(clone_spec, encryption_type, clone_passphrase):
+            log.error(f"Apply RBD encryption {encryption_type} failed on {image_spec}")
+
+        encryption_config.append({encryption_type: clone_passphrase})
+        out, err = rbd.device_map(clone_spec, "nbd", reversed(encryption_config))
+        log.info(f"Out, Err : {out}: {err}")
+        if err:
+            log.error(
+                f"RBD device map failed for {image_spec} and encryption_config: {encryption_config}"
+            )
+
+        kw["config"]["device_names"].append(out)
+        kw["config"]["image_spec"].append(clone_spec)
+        krbd_io_handler(**kw)
+
+        flatten_config = {"encryption_config": reversed(encryption_config)}
+        if rbd.flatten_clone(pool, clone, **flatten_config):
+            log.error(f"Flatten clone with encryption failed for {clone_spec}")
+
+        log.info(
+            f"Verifying data integrity between original image: {image_spec} and clone: {clone_spec}"
+        )
+        check_data(rbd, image_spec, clone_spec)
+
+    except RbdBaseException as error:
+        print(error.message)
+        return 1
+
+
+def cleanup(rbd_obj, **kw):
+    """Cleanup the replicated and ecpools created for the test module
+
+    Args:
+        rbd_obj (dict): dictionary containing rbd ecpool and replicated pool object
+    """
+    log.info("Cleaning up pools")
+    obj = rbd_obj.get("rbd_ecpool", rbd_obj["rbd_reppool"])
+    obj.clean_up(
+        pools=[
+            kw["config"].get("rep_pool_config", {})["pool"],
+            kw["config"].get("ec_pool_config", {})["pool"],
+        ]
+    )
+
+
+def run(**kw):
+    """Verify the luks encryption functionality for image and clone
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common, fio and rbd-nbd packages,
+    conf and keyring files
+
+    Test case covered -
+    1) CEPH-83575263 - Encrypt & decrypt file using same keys and different keys
+    Test Case Flow
+    1. Create a pool and an Image
+    2. Apply Encryption LUKS1 format to the clone image
+    3. Perform resize operations to compensate the overhead due to LUKS1 header
+    4. Load encryption using rbd device map
+    5. Mount the image and run IOs
+    6. Create snapshot, Protect the snapshot and run IOs
+    7. Perform clone from the snapshot of the image created
+    8. Apply encryption with different passphrase key other than parent image
+    9. Load the encryption using rbd device map on cloned image
+    10. Mount the clone and run Ios
+    11. Flatten the clone
+    12. Verify data integrity between mounted clone image and the original image
+    """
+    log.info("Running Trash function")
```

Comment:
can we mention polarion id and test description here, I don't think it's Trash function execution

Reply:
Changed this to running luks encryption tests. Can't mention polarion ID since this test module is generic for two polarion test cases.